Ethernet on ATMEGA4809 still not working

Dear Forum,
I am trying to get ethernet working on a ATMEGA4809 40 pin, a W5500 module is attached to the SPI interface (See attachments)
If I compile code for Nano Every, which also has ATMEGA4809 processor it just passes without errors.
But if I compile code for ATMEGA4809 the compiler stops with error:
c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp:99:20: error: 'INADDR_NONE' was not declared in this scope 

For code and and errorlog see attachments.

I hope you can help me with this,
Werner

[Errorlog_4809_ethernet.txt](https://github.com/user-attachments/files/19774779/Errorlog_4809_ethernet.txt)
[Testcode_4809_ethernet.txt](https://github.com/user-attachments/files/19774777/Testcode_4809_ethernet.txt)
[ATMEGA4809_W5500_SD.pdf](https://github.com/user-attachments/files/19774778/ATMEGA4809_W5500_SD.pdf)

Dear Hans,

 

Thank you so much for your email. I was so happy to see it, I thought my problem is solved!

 

I have updated to 1.1.4, but unfortunately the problem is still the same. I also did a reinstall of ethernet and MegaCoreX.

(c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp:99:20: error: 'INADDR_NONE' was not declared in this scope)   

Could you please have a look at it otherwise I have to redesign a complete PCB (Big one ☹) and use other processor.

 

Attached the errorlog and code snippet.

 

Thank you very much in advance,

 

Werner

 

 

Met vriendelijke groeten,

 

Ing. Werner F.P.Lösken

 

Eendrachtstraat 12

B-3920 Lommel

Belgium

Gsm:  +32 (0) 498 84 00 78

Email: ***@***.*** ***@***.***> 

P Please consider the environment before printing this e-mail

 

 

From: Hans ***@***.***> 
Sent: woensdag 14 mei 2025 19:47
To: MCUdude/MegaCoreX ***@***.***>
Cc: Werner-Losken ***@***.***>; Author ***@***.***>
Subject: Re: [MCUdude/MegaCoreX] Ethernet on ATMEGA4809 still not working (Issue #207)

 

  <https://avatars.githubusercontent.com/u/3238730?s=20&v=4> MCUdude left a comment (MCUdude/MegaCoreX#207) <https://github.com/MCUdude/MegaCoreX/issues/207#issuecomment-2881052908> 

v1.1.4 is now available through boards manager, which should have this issue resolved.

—
Reply to this email directly, view it on GitHub <https://github.com/MCUdude/MegaCoreX/issues/207#issuecomment-2881052908> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/BNQEUC3FC2BWWKONI52BWJL26N6Q3AVCNFSM6AAAAAB3HZ67D2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOBRGA2TEOJQHA> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/BNQEUC6U2KJT2SNRLWPZXAT26N6Q3A5CNFSM6AAAAAB3HZ67D2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTVLXFQOY.gif> Message ID: ***@***.*** ***@***.***> >


#include <SPI.h>
#include <Ethernet.h>


byte mac[] = { 0xA8, 0x61, 0x0A, 0xAE, 0x79, 0xF9 };  
IPAddress ip(192,168,1,170);

void setup() 
{
SPI.begin();
Serial.begin(57600,SERIAL_8N1);
}

void loop()
{
  Serial.println("Starting w5500");
  Ethernet.begin(mac,ip);
  Serial.println(Ethernet.localIP());
  while(1){}
}


FQBN: MegaCoreX:megaavr:4809:pinout=40pin_standard,resetpin=gpio
Using board '4809' from platform in folder: C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4
Using core 'coreX-corefiles' from platform in folder: C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4

Detecting libraries used...
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard C:\Users\WinAVR\AppData\Local\arduino\sketches\C355C6BF761CB56002504C781CD59006\sketch\Internet_4809.ino.cpp -o nul
Alternatives for SPI.h: ***@***.***
ResolveLibrary(SPI.h)
  -> candidates: ***@***.***
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src C:\Users\WinAVR\AppData\Local\arduino\sketches\C355C6BF761CB56002504C781CD59006\sketch\Internet_4809.ino.cpp -o nul
Alternatives for Ethernet.h: ***@***.***
ResolveLibrary(Ethernet.h)
  -> candidates: ***@***.***
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src C:\Users\WinAVR\AppData\Local\arduino\sketches\C355C6BF761CB56002504C781CD59006\sketch\Internet_4809.ino.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src\SPI.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dhcp.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Ethernet.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\EthernetClient.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\EthernetServer.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\EthernetUdp.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\socket.cpp -o nul
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\utility\w5100.cpp -o nul
Generating function prototypes...
C:\Users\WinAVR\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\7.3.0-atmel3.6.1-arduino7/bin/avr-g++ -c -g -Os -w -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -flto -mrelax -w -x c++ -E -CC -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/deprecated -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\variants\40pin-standard -IC:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI\src -Ic:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src C:\Users\WinAVR\AppData\Local\arduino\sketches\C355C6BF761CB56002504C781CD59006\sketch\Internet_4809.ino.cpp -o C:\Users\WinAVR\AppData\Local\Temp\1375027725\sketch_merged.cpp
C:\Users\WinAVR\AppData\Local\Arduino15\packages\builtin\tools\ctags\5.8-arduino11/ctags -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives C:\Users\WinAVR\AppData\Local\Temp\1375027725\sketch_merged.cpp

Compiling sketch...
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\sketch\\Internet_4809.ino.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\sketch\\Internet_4809.ino.cpp.o"
Compiling libraries...
Compiling library "SPI"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src\\SPI.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\SPI\\SPI.cpp.o"
Compiling library "Ethernet"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\Dhcp.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\Dhcp.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\EthernetServer.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\EthernetServer.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\EthernetClient.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\EthernetClient.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\EthernetUdp.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\EthernetUdp.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\Dns.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\Dns.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\Ethernet.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\Ethernet.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\socket.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\socket.cpp.o"
"C:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\avr-gcc\\7.3.0-atmel3.6.1-arduino7/bin/avr-g++" -c -g -Os -std=gnu++17 -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -Wno-error=narrowing -MMD -flto -mrelax -mmcu=atmega4809 -DF_CPU=16000000L -DARDUINO=10607 -DARDUINO_AVR_ATmega4809 -DARDUINO_ARCH_MEGAAVR "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles/api/deprecated" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\cores\\coreX-corefiles" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\variants\\40pin-standard" "-IC:\\Users\\WinAVR\\AppData\\Local\\Arduino15\\packages\\MegaCoreX\\hardware\\megaavr\\1.1.4\\libraries\\SPI\\src" "-Ic:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src" "c:\\Users\\WinAVR\\Documents\\Arduino_2.01\\libraries\\Ethernet\\src\\utility\\w5100.cpp" -o "C:\\Users\\WinAVR\\AppData\\Local\\arduino\\sketches\\C355C6BF761CB56002504C781CD59006\\libraries\\Ethernet\\utility\\w5100.cpp.o"
c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp: In member function 'int DNSClient::getHostByName(const char*, arduino::IPAddress&, uint16_t)':
c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp:99:20: error: 'INADDR_NONE' was not declared in this scope
  if (iDNSServer == INADDR_NONE) {
                    ^~~~~~~~~~~
c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp:99:20: note: suggested alternative:
In file included from C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/Client.h:22:0,
                 from C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/ArduinoAPI.h:29,
                 from C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/Arduino.h:23,
                 from c:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet\src\Dns.cpp:5:
C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\cores\coreX-corefiles/api/IPAddress.h:83:24: note:   'arduino::INADDR_NONE'
 extern const IPAddress INADDR_NONE;
                        ^~~~~~~~~~~
Using library SPI at version 1.0 in folder: C:\Users\WinAVR\AppData\Local\Arduino15\packages\MegaCoreX\hardware\megaavr\1.1.4\libraries\SPI 
Using library Ethernet at version 2.0.2 in folder: C:\Users\WinAVR\Documents\Arduino_2.01\libraries\Ethernet 
exit status 1

Compilation error: exit status 1

--- a/megaavr/cores/coreX-corefiles/api/IPAddress.h
+++ b/megaavr/cores/coreX-corefiles/api/IPAddress.h
@@ -80,6 +80,7 @@ class IPAddress : public Printable
   friend ::DNSClient;
 };
 
-extern const IPAddress INADDR_NONE;
 }
 using arduino::IPAddress;
+
+extern const IPAddress INADDR_NONE;

